### PR TITLE
NVFuser fixes

### PR DESCRIPTION
**Pull request by MaximumEntropy**

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -60,6 +60,9 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             seed=self.cfg.get('seed', 1234),
         )
 
+        if not self.cfg.get('fused_bf16'):
+            set_jit_fusion_options()
+
         if isinstance(self.trainer.accelerator_connector.precision_plugin, NLPNativeBfloat16PrecisionPlugin):
```
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:66]: We can remove this?

File: nemo/collections/nlp/modules/common/megatron/utils.py [hunk 2]
```diff
@@ -164,11 +164,3 @@ def autocast_forward(self, *args):
...
             else:
                 return self.func(*args)
-
-
-@contextmanager
-def dummy_handler():
-    try:
-        yield
-    finally:
-        pass
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/megatron/utils.py:174]: Don't we use this?

Comment (MaximumEntropy) [reply nemo/collections/nlp/modules/common/megatron/utils.py:174]: No longer, I had missed removing it in some other places and just fixed it.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 1 above)
Comment (MaximumEntropy) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:66]: Remove it entirely? I think we want to disable jit fusion for bf16 right?

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**